utility/utility.go

```diff
@@ -125,23 +125,47 @@ func GetSubdirectoryRelativePath(subdirectoryPath string, directoryPath string)
 	return NormalizePath(SanitizePath(strings.TrimPrefix(subdirectoryPath, directoryPath)))
 }
 
-var copyBufPool = sync.Pool{New: func() interface{} {
-	return make([]byte, CopiedBlockMaxSize)
-}}
+// BytesPool holds []byte.
+type BytesPool struct {
+	pool chan []byte
+}
 
-func getCopyBuf() []byte {
-	return copyBufPool.Get().([]byte)
+// NewBytesPool creates a new pool of Clients.
+func NewBytesPool(max int) *BytesPool {
+	return &BytesPool{
+		pool: make(chan []byte, max),
+	}
 }
 
-func putCopyBuf(buf []byte) {
-	copyBufPool.Put(buf)
+const CopyBytesPoolSize = 2
```

Comment:
We don't have convenient way to set size of the pool, so just provide default for now.